swing/src/net/sf/openrocket/gui/scalefigure/RocketPanel.java

```diff
@@ -606,20 +606,8 @@ private void updateExtras() {
 		figure3d.setCG(cg);
 		figure3d.setCP(cp);
 
-		// Length bound is assumed to be tight
-		double length = 0;
-		Collection<Coordinate> bounds = curConfig.getBounds();
-		if (!bounds.isEmpty()) {
-			double minX = Double.POSITIVE_INFINITY, maxX = Double.NEGATIVE_INFINITY;
-			for (Coordinate c : bounds) {
-				if (c.x < minX)
-					minX = c.x;
-				if (c.x > maxX)
-					maxX = c.x;
-			}
-			length = maxX - minX;
-		}
-
+		double length = curConfig.getLength();
```

Comment:
I think we want to actually preserve the previous behavior, here.  -- see the comment about increasing fin-height.  (for example)

Reply:
That's not actually causing the fin-height scaling issue, but the fin height-issue is a good catch. Using the old method in here essentially just recalculates the value in essentially the same way that has already been calculated so it feels like just extra processing. As such, I'd prefer to keep this change here and fix the fin-height scaling issue.

It took me a few minutes to figure out what was going on with the fin-height issue, but its pretty clear and straightforward. The fin-height issue is that I'm not properly calculating the max height the fins will span in all the way around the rocket. Using the approach I used for the FinSets, you get the tightest bounds for the fins in the side profile.

Fortunately, the FinSet already has a getBoundingBox() method which _does_ take this into account. The FinSet.getBoundingBox() method sets the bounds for its bounding box to be the length of the fins x 2 * (finHeight - 0).